File: gerdo.java
import java.util.Scanner;

public class gerdo {

   public static void main(String[] args) {
       Scanner scanner = new Scanner(System.in);

       int n = scanner.nextInt(); // Distance to the wall
       int x = scanner.nextInt(); // Length of the foot
       int y = scanner.nextInt(); // Width of the foot

       int[] result = solveGame(n, x, y);

       if (result[0] == -1) {
           System.out.println("-1");
       } else {
           System.out.println( result[0]);
           System.out.println( result[1]);
       }
   }

   public static int[] solveGame(int n, int x, int y) {
       int maxLongitudinalSteps = n / x;
       int maxTransverseSteps = n / y;

       if (n % x != n % y) {
           return new int[]{-1}; // No solution
       }

       int longitudinalSteps = 0;
       while ((n - (longitudinalSteps * x)) % y != 0) {
           longitudinalSteps++;
       }

       int transverseSteps = n - (longitudinalSteps * x);

       return new int[]{longitudinalSteps, transverseSteps};
   }
}

File: gerdo.py
def solve_game(n, x, y):
  """
  حل بازی "گردو، شکستم" یک نفره

  Args:
      n: فاصله مهدی تا دیوار (سانتی‌متر)
      x: طول پای مهدی (سانتی‌متر)
      y: عرض پای مهدی (سانتی‌متر)

  Returns:
      (طولی, عرضی): تعداد دفعاتی که مهدی باید پایش را به صورت طولی و عرضی بگذارد.
      -1: اگر راه حلی برای رسیدن به دیوار وجود نداشته باشد.
  """

  max_longitudinal_steps = n // x
  max_transverse_steps = n // y

  if n % x != n % y:
    return -1

  longitudinal_steps = 0
  while (n - (longitudinal_steps * x)) % y != 0:
    longitudinal_steps += 1

  transverse_steps = n - (longitudinal_steps * x)

  return (longitudinal_steps, transverse_steps)

# مثال
n = 100
x = 20
y = 15

result = solve_game(n, x, y)

if result == -1:
  print("راه حلی برای رسیدن به دیوار وجود ندارد.")
else:
  longitudinal_steps, transverse_steps = result
  print(f"طولی: {longitudinal_steps}")
  print(f"عرضی: {transverse_steps}")
